docs/architecture.md

```diff
@@ -0,0 +1,223 @@
+# Architecture
+
+The core of OSP-Magnum is best described as a library that provides a set of functions and data structures intended to support a space flight simulator. The user is responsible for implementing their own application on the top level, allowing for full control over every feature.
+
+OSP-Magnum provides two distinct worlds the application can manage:
+* **Universe**: A space simulator that can handle massive world sizes with multiple coordinate spaces. Objects in space are referred to as **Satellites**.
+* **ActiveScene**: A typical game engine that can support physics, rendering, etc..
+
+The ActiveScene can be synchronized to represent a small area of the Universe, where in-game representations of nearby Satellites can be added to the scene. This area is referred to as the **ActiveArea** and can follow the user around to seamlessly explore the Universe while also avoiding floating point precision errors.
+
+For performance and flexibility for such a project, we adhere to few techniques that may seem unconventional:
+* Data-oriented Design
+* Entity Component System
+
+## Data-oriented Design (DoD)
+
+Data-oriented Design is an emerging approach to software architecture, used extensively in OSP-Magnum. By focusing on creating a memory layout for efficient access, DoD greatly reduces complexity and achieves high performance. As counterintuitive as it may seem, this potentially results in (or at least encourages) a greater degree of flexibility compared to class-based OOP.
```

Comment:
I wouldn't say DoD necessarily reduces complexity, that's more of the ECS/composition part which isn't necessarily implemented in a DoD-manner. Also, I'd suggest

"By building the engine architecture around a memory layout which facilitates efficient data access..."

Reply:
DoD is well known for reducing complexity. I'm too lazy to get specific links at the moment, but many talks and articles mention this. Other approaches tend to hide complexity under cleaner syntax, automatic memory management, etc... It still is much simpler compared to C++'s normal OOP.